### ISSUE TITLE
Ionic V4.beta11 ion-back-button changes tabs

### ISSUE THREAD

**Issue by gfitas** [ionic-team/ionic#15494]
# Bug Report

**Ionic Info**

```
ionic (Ionic CLI)          : 4.1.2  (/usr/local/lib/node_modules/ionic)
Ionic Framework            : @ionic/angular 4.0.0-beta.8
@angular-devkit/core       : 0.7.5
@angular-devkit/schematics : 0.7.5
@angular/cli               : 6.1.5
@ionic/ng-toolkit          : 1.0.8
@ionic/schematics-angular  : 1.0.6
```

**Describe the Bug**
I have a tab page with 3 differents page. In the second tab, you can push another page. But when you want to come back to the tab page thanks to the ion-back-button, you come back to the first tab instead of the second.

Thank you in advance

**Comment by siva563** [comment]
@gfitas  did you already implemented tabs could you please help me.

**Comment by yuankunluo** [comment]
@gfitas , i can help. i have done some tabs with Ionic V4

**Comment by siva563** [comment]
@yuankunluo Could you please help me for Tabs Issue.

**tabs.html**
```
  <ion-tabs color="primary" class="activity-tabs" tabbarPlacement="top" tabbar-highlight="true" translucent="true" use-router="true">
    <ion-tab padding-horizotal class="ionTabsPublic" label="Public" href="activeTabs/tabs/(public:public)">
      <ion-router-outlet stack name="public"></ion-router-outlet>
    </ion-tab>
    <ion-tab padding-horizotal label="Private" href="activeTabs/tabs/(private:private)">
      <ion-router-outlet stack name="private"></ion-router-outlet>
    </ion-tab>
  </ion-tabs>
```

**tabs.module.ts**
```

const routes: Routes = [
  {
    path: 'tabs',
    component: activeTabsPage,
    children: [
      {
        path: 'public',
        outlet: 'public',
        component: PublicPage
      },
      {
        path: 'private',
        outlet: 'private',
        component: PrivatePage
      }
    ]
  },
  {
    path: '',
    redirectTo: '/activeTabs/tabs/(public:public)'
  }
];
```

I am facing the problem `PrivatePage` and `PublicPage` components are loaded only once, if I need fetch the data from serve dynamically click on tab. I mean when I am clicking on tab data is not refreshing I wrote the code into `ngOninit() `, but it is calling only once. could you please help me solve this issue

**Comment by gfitas** [comment]
Still in V4.beta11

**Comment by hghammoud** [comment]
I have the same issue. 

It's is related to the route at which the back button is redirecting. its redirecting to /tabs instead of /tabs/(outlet:outletname). Seems that the history is not storing the outlet

**Comment by wamburu** [comment]
Experiencing same issue with  Beta 12.

However, using [`NavController.pop()`]( https://beta.ionicframework.com/docs/api/nav-pop) works just fine.

**Comment by guigiani** [comment]
Same issue here:

   ionic (Ionic CLI)             : 4.3.1 (C:\Users\<user>\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework               : @ionic/angular 4.0.0-beta.15
   @angular-devkit/build-angular : 0.8.6
   @angular-devkit/schematics    : 0.8.6
   @angular/cli                  : 6.2.6
   @ionic/angular-toolkit        : 1.1.0

**Comment by tomfebry** [comment]
Still an issue in beta 16

**Comment by HankXu** [comment]
I have the same issue. And I navigate with the `NavController.goBack()` now, it can really back to the latest selected tab.

**Comment by janilamppu** [comment]
Confirmed.

`Ionic Framework: @ionic/angular 4.0.0-beta.17`

**Comment by janilamppu** [comment]
I updated my project to beta 19 and the issue still persists. I have 3 tabs and a button on the second tab to navigate to a new page. When Back-button is clicked in the new page, I get thrown to tab 1 instead of tab 2. Using defaultHref makes no change. I also changed the page navigation to use routerLink instead of href as adviced in the changelogs.

`"@ionic/angular": "^4.0.0-beta.19",`

**Comment by tomfebry** [comment]
Still seems to be in an issue in beta 19

**Comment by HankXu** [comment]
It still an issue in beta 19, though you can hack it. [#15216](https://github.com/ionic-team/ionic/issues/15216)

**Comment by yuricamara** [comment]
Still an issue in 4.0.0-rc.0

For example,
`<ion-back-button defaultHref="tabs/notifications"></ion-back-button>`
doesn´t work as expected.

**Comment by bossjon** [comment]
+1
Still an issue in 4.0.0-rc.1

**Comment by manucorporat** [comment]
Can someone explain better what the actual issue is?

**Comment by manucorporat** [comment]
@NiklasMerz right! i am confused with the @bossjon comment, saying that this is still a problem in rc.1

I have the feeling some people are talking about different issues, which makes things even more confusing? 

@NiklasMerz  Do you think it's good to close? We can always open a new issue

**Comment by NiklasMerz** [comment]
I think its good to be closed. The discussion here is to confusing here anyway. If someone still has this problem or something similar a new issue might be better.

**Comment by bossjon** [comment]
@manucorporat @NiklasMerz 
ok,
1. `ionic start tabIssue tabs --type=angular`
2. `ionic generate page new-page`
3. `tab2.html`:
```
<ion-content padding>
  <ion-button color="primary" [routerLink]="[ '/new-page' ]" routerDirection="forward">
    Open new page
  </ion-button>
</ion-content>
```
Open `tab2` page and navigate to a new-page. When back-button is clicked in the new-page, you get thrown to `tab1` instead of `tab2`. 

Ps: If you refresh the `new-page` with `defaultHref="/tabs/tab2"`, then press back-button works fine.

**Comment by NiklasMerz** [comment]
OK I am navigating with the `NavController` in my class that`s why it works for me.

**Comment by bossjon** [comment]
@NiklasMerz 
html:
```
  <ion-button color="primary" (click)="go()">
    Open new page
  </ion-button>
```
ts: 
```
  go() {
    this.nav.navigateForward('new-page');
  }
```
not working, can you share your code?

**Comment by NiklasMerz** [comment]
Ok my bad. Yes your description is correct. I can confirm this issue with a new project.

(I forgot that I replaced all actual `ion-back-button`s with my own button which navigates to the predefined page manually. Sorry for the confusion, stupid mistake, I should have looked carefully)

**Comment by paulstelzer** [comment]
I close this as duplicate of #15216

I think my gif demonstrate the issue well -> https://github.com/ionic-team/ionic/issues/15216#issuecomment-447334318 (after navigating back it opens the first tab)

and I think it's the same issue like described here
